diffkemp/semdiff/result.py

```diff
@@ -37,11 +37,49 @@ def __init__(self, name, filename=None, callstack=None):
             self.filename = filename
             self.callstack = callstack
 
+    class MacroElement:
+        """
+        Represents a macro element on the macro stack.
+        """
+        def __init__(self, yaml):
+            """Transforms YAML to MacroElement object."""
+            self.name = yaml["name"]
+            self.body = yaml["body"]
+            self.file = yaml["file"]
+            self.line = yaml["line"]
+
+    class MacroDifference:
+        """
+        Represents a macro difference as found by SimpLL.
+        """
+        def __init__(self, yaml):
+            """Transforms YAML to MacroDifference object."""
+            self.name = yaml["name"]
+            self.first_stack = map(Result.MacroElement, yaml["left-stack"])
+            self.second_stack = map(Result.MacroElement, yaml["right-stack"])
+            self.first_line = yaml["left-line"]
+            self.second_line = yaml["right-line"]
+
+        def __str__(self):
+            res = "Macro " + self.first_stack[0].name + " is different:\n"
+            for stack in [self.first_stack, self.second_stack]:
+                for macro in stack:
+                    res = res + "  from macro " + macro.name + " in file " + \
+                        macro.file + " on line " + str(macro.line) + "\n"
+                res = res + "  used in the function on line " + \
+                    (str(self.first_line) if stack is self.first_stack else
+                     str(self.second_line)) + "\n"
+                res = res + "    " + stack[0].body + "\n"
+                if stack is self.first_stack:
+                    res = res + "\n"
```

Comment:
I think using the style "Some {} string".format(x) is better readable than using "Some " + x + " string".